packages/core/src/core.ts

```diff
@@ -282,19 +288,17 @@ function handleTypeCasting(
         attributeToAdd = attributeToAdd.bind(this)
       }
       break
-    case Number:
+    case "number":
       attributeToAdd = parseFloat(attributeToAdd)
       break
-    case Boolean:
+    case "boolean":
       attributeToAdd = /^[ty1-9]/i.test(attributeToAdd)
       break
-    case Object:
+    case "array":
+    case "object":
```

Comment:
I think we should fully remove both of these and just have a new one called "json" since that's really what it's doing.

Reply:
done